common/src/main/java/org/opensearch/ml/common/input/parameter/regression/LogisticRegressionParams.java

```diff
@@ -0,0 +1,244 @@
+                    break;
+                case TARGET_FIELD:
+                    target = parser.text();
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+        return new LogisticRegressionParams(objective,  optimizerType,  learningRate, epsilon, epochs, batchSize, loggingInterval, seed, target);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (objectiveType != null) {
+            out.writeBoolean(true);
+            out.writeEnum(objectiveType);
+        } else {
+            out.writeBoolean(false);
+        }
+        if (optimizerType != null) {
+            out.writeBoolean(true);
+            out.writeEnum(optimizerType);
+        } else {
+            out.writeBoolean(false);
+        }
+        out.writeOptionalDouble(learningRate);
+        out.writeOptionalDouble(epsilon);
+        out.writeOptionalInt(epochs);
+        out.writeOptionalInt(batchSize);
+        out.writeOptionalInt(loggingInterval);
+        out.writeOptionalLong(seed);
+        out.writeOptionalString(target);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (objectiveType != null) {
+            builder.field(OBJECTIVE_FIELD, objectiveType);
+        }
+        if (optimizerType != null) {
+            builder.field(OPTIMISER_FIELD, optimizerType);
+        }
+        if (learningRate != null) {
+            builder.field(LEARNING_RATE_FIELD, learningRate);
+        }
+        if (epsilon != null) {
+            builder.field(EPSILON_FIELD, epsilon);
+        }
+        if (epochs != null) {
+            builder.field(EPOCHS_FIELD, epochs);
+        }
+        if (batchSize != null) {
+            builder.field(BATCH_SIZE_FIELD, batchSize);
+        }
+        if (loggingInterval != null) {
+            builder.field(LOGGING_INTERVAL_FIELD, loggingInterval);
+        }
+        if (seed != null) {
+            builder.field(SEED_FIELD, seed);
+        }
+        if (target != null) {
+            builder.field(TARGET_FIELD, target);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return PARSE_FIELD_NAME;
+    }
+
+    @Override
+    public int getVersion() {
+        return 1;
+    }
+
+    public enum ObjectiveType {
+        HINGE,
+        LOGMULTICLASS;
+        public static ObjectiveType from(String value) {
+            try{
+                return ObjectiveType.valueOf(value);
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Wrong objective type");
+            }
+        }
+    }
+
+    public enum OptimizerType {
+        SIMPLE_SGD,
+        LINEAR_DECAY_SGD,
+        SQRT_DECAY_SGD,
+        ADA_GRAD,
```

Comment:
@Craigacp help review if all these supported for logistic regression?

Reply:
These are all supported.